xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1640,6 +1644,24 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
         st->colorTransferCharacteristic = pStream->codecpar->color_trc;
         st->colorRange = pStream->codecpar->color_range;
 
+        if (pStream->side_data && (pStream->side_data->type == AV_PKT_DATA_DOVI_CONF))
+          st->hdr_type = StreamHdrType::HDR_TYPE_DOLBYVISION;
+        else if (st->colorPrimaries == AVCOL_PRI_BT2020)
+        {
+          if (st->colorTransferCharacteristic == AVCOL_TRC_ARIB_STD_B67) // hlg
+            st->hdr_type = StreamHdrType::HDR_TYPE_HLG;
+          else if (st->colorTransferCharacteristic == AVCOL_TRC_SMPTEST2084) // hdr 10 bit
```

Comment:
`AVCOL_TRC_SMPTEST2084` seems is deprecated and in any case has always same value that `AVCOL_TRC_SMPTE2084`. See https://ffmpeg.org/doxygen/trunk/pixfmt_8h_source.html#l00473 then can not be used to differentiate HDR type.

Also I am not aware that there may be any 12 bit HDR content other than DolbyVision (which would be identified as HDR_TYPE_DOLBYVISION), so I do not see the need for the "HDR12" tag.

I propose to simplify this block like this:

```c++
        if (pStream->side_data && (pStream->side_data->type == AV_PKT_DATA_DOVI_CONF))
        {
          st->hdr_type = StreamHdrType::HDR_TYPE_DOLBYVISION;
        }
        else if (st->colorPrimaries == AVCOL_PRI_BT2020)
        {
          if (st->colorTransferCharacteristic == AVCOL_TRC_SMPTE2084) // hdr10
            st->hdr_type = StreamHdrType::HDR_TYPE_HDR10;
          else if (st->colorTransferCharacteristic == AVCOL_TRC_ARIB_STD_B67) // hlg
            st->hdr_type = StreamHdrType::HDR_TYPE_HLG;
        }
```

Reply:
I concur with @thexai in the belief that the only spec that allows for 12 bit content is DV, however even though it's in the spec I'm not aware of any content that uses 12 bit yet. Mostly likely 12 bit content is seen as pointless currently in the consumer space as consumer displays can do no more than 10 bit.

Reply:
Yes, I'm aware that no consumer displays do more than 10 bit (yet), I was just trying to be a little "future proof" in adding 12 bit and associated flag.  FWIW, the HDR10+ spec is 10 bit or better to a maximum of 16 bits.  Anyway, I'm quite happy to drop the 12 bit stuff, can always be added at a later date should there be a need.

Reply:
I was sure I'd read HDR10+ was limited to 10 bit only, hence the name where + only refers to dynamic as apposed to static metadata. Would be interested if you've seen something different written somewhere.